### PR TITLE
Feature/zip over lists of streams

### PR DESCRIPTION
**Pull request by rkraneis**
Hi, I've added a function Stream::zip to zip Lists of Streams with a combiner working on Lists of values.

### CODE REVIEW

**Comment by poetix** [comment]
This seems similar in functionality to what `StreamUtils::merge` and `MergingSpliterator` do - slightly more general, as it accepts a `Function<List<T>, O>` rather than a reducer. I wonder whether the common behaviour could be factored out?

**Comment by rkraneis** [comment]
I've brought the functionality even more in line with MergingSpliterator, but I think the remaining differences (list <> reduction, truncation to shortest stream <> ignorance of missing elements) are quite fundamental and would lead to awkward code.

File: src/main/java/com/codepoetics/protonpack/ListZippingSpliterator.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                .mapToLong(Spliterator::estimateSize)
+                .min().orElse(0);
+    }
+
+    @Override
+    public int characteristics() {
+        // TODO: benchmark and cache?
+        int characteristics = spliterators.stream()
+                .mapToInt(Spliterator::characteristics)
+                .reduce(0, (i, j) -> i & j);
```
Comment (poetix) [inline src/main/java/com/codepoetics/protonpack/ListZippingSpliterator.java]: Do this in the factory method.

File: src/main/java/com/codepoetics/protonpack/ListZippingSpliterator.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public Spliterator<O> trySplit() {
+        return null;
+    }
+
+    @Override
+    public long estimateSize() {
+        // TODO: benchmark and cache?
+        return spliterators.stream()
+                .mapToLong(Spliterator::estimateSize)
+                .min().orElse(0);
```
Comment (poetix) [inline src/main/java/com/codepoetics/protonpack/ListZippingSpliterator.java]: Do this in the factory method.

**Comment by poetix** [comment]
Merged.

**Pull request merged**